examples/gno.land/r/nt/boards2/v1/freeze.gno

```diff
@@ -0,0 +1,76 @@
+package boards2
+
+import "std"
+
+type freezeArgs struct {
+	boardID  BoardID
+	threadID PostID
+	replyID  PostID
+}
+
+func (args freezeArgs) getThread() (*Post, *Board) {
+	board := mustGetBoard(args.boardID)
+	assertBoardIsNotFrozen(board)
+
+	thread := mustGetThread(board, args.threadID)
+	return thread, board
+}
+
+func (args freezeArgs) getReply() (*Post, *Board) {
+	board := mustGetBoard(args.boardID)
+	assertBoardIsNotFrozen(board)
+
+	thread := mustGetThread(board, args.threadID)
+	assertThreadVisible(thread)
+	assertThreadIsNotFrozen(thread)
+
+	reply := mustGetReply(thread, args.replyID)
+	assertReplyVisible(reply)
+	return reply, board
+}
+
+func setReplyReadOnly(args freezeArgs, isReadOnly bool) {
+	reply, board := args.getReply()
+
+	if isReadOnly {
+		// disallow freezing already frozen reply
+		assertReplyIsNotFrozen(reply)
+	}
+
+	caller := std.OriginCaller()
```

Comment:
```suggestion
	caller := std.PreviousRealm().Address()
```

https://github.com/gnolang/gno/blob/master/docs/resources/effective-gno.md#contract-level-access-control

Reply:
Addressed in 9c57c0b